eng/Directory.Build.Common.targets

```diff
@@ -27,7 +27,7 @@
   </Target>
 
   <!-- Validates that there is a release note entry for the current package version -->
-  <Target Name="ValidateReleaseNotes" BeforeTargets="SetPackageProjectUrlandReleaseNotes" Condition="'$(IsShippingLibrary)' == 'true'" >
+  <!-- <Target Name="ValidateReleaseNotes" BeforeTargets="SetPackageProjectUrlandReleaseNotes" Condition="'$(IsShippingLibrary)' == 'true'" >
```

Comment:
why did we need to exclude this?

Reply:
If we leave it in, we will be required to add release notes before being able to create a NuGet package with different version number.  An example might help explain this better.

Right now, when we build a NuGet package it will be versioned like this: `Azure.Messaging.ServiceBus.7.16.0-alpha.20230620.1.nupkg`. We get the same version every time we build in the same day.

If we needed to release a patch to `7.16.0` we would need to do it on another day, so the `alpha.YYYYMMDD` would be different or version bump to `7.16.1`.

I felt that it was an odd restriction to place on ourselves to have to write release notes for such a change.

Reply:
I think we should add a quick blurb in release notes :)

Reply:
Reenabled release note validation and added the blurb!